### PR TITLE
Don't let DataTuple inherit from TestTuple

### PR DESCRIPTION
**Pull request by thomkeh**
And also introduce the new `LabelTuple`.

The names are all subject to change. I can easily rename them with my editor.

Myles might be excited about this:

```python
preds = Prediction(hard=pd.Series([0, 0, 1, 1]))
targets = LabelTuple.from_np(
    s=np.array([0, 1, 1, 1]), y=np.array([0, 1, 0, 1]), s_name="s", y_name="y"
)
score = metric.score(preds, targets)
```

### CODE REVIEW

File: ethicml/utility/data_structures.py [hunk 1]
```diff
@@ -256,6 +287,150 @@ def from_npz(cls, data_path: Path) -> DataTuple:
...
+        assert len(s) == len(y), "data has to have the same length"
+        return cls(
+            data=pd.concat([s, y], axis="columns", sort=False),  # type: ignore[arg-type]
+            s_column=s_column,
+            y_column=y_column,
+            name=name,
+        )
+
+    @classmethod
+    def from_np(cls, *, s: npt.NDArray, y: npt.NDArray, s_name: str, y_name: str) -> LabelTuple:
```
Comment (MylesBartlett) [inline ethicml/utility/data_structures.py:321]: Maybe make "s" and "y" the default names?

Comment (thomkeh) [reply ethicml/utility/data_structures.py:321]: sure, good idea

File: ethicml/algorithms/inprocess/fair_dummies.py [hunk 2]
```diff
@@ -52,7 +52,9 @@ def _get_path_to_script(self) -> List[str]:
 
     @implements(InAlgorithmSubprocess)
     def _get_flags(self) -> FairDummiesArgs:
-        model_type = "deep_model" if self.model_type.lower() == "deep_model" else "linear_model"
+        model_type: Literal["deep_model", "linear_model"] = (
+            "deep_model" if self.model_type.lower() == "deep_model" else "linear_model"
+        )
```
Comment (olliethomas) [inline ethicml/algorithms/inprocess/fair_dummies.py:57]: BTW, this is only here to satisfy the Literal type. For future work, the Literal should be replaced with an Enum.

**Review by olliethomas — APPROVED** [review]

File: ethicml/algorithms/inprocess/fair_dummies.py (hunk 2 above)
Comment (thomkeh) [reply ethicml/algorithms/inprocess/fair_dummies.py:57]: should be easy to do

**Pull request merged**